lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -187,6 +188,17 @@ def add_payment_method(post, payment_method, options)
         end
       end
 
+      def add_network_token_fields(post, payment_method)
+        if payment_method.source == :apple_pay || payment_method.source == :google_pay
+          post[:cavv] = payment_method.payment_cryptogram
+          post[:eci] = payment_method.eci
+          post[:decrypted_applepay_data] = 1
+          post[:decrypted_googlepay_data] = 1
```

Comment:
Do both of these need to be set to `1` regardless of which payment method type is being used?

Reply:
Hi @naashton base on the NMI docs yes, for both cases set 1 to indicate decrypted data is being sent.